src/ansys/fluent/core/session_solver.py

```diff
@@ -228,3 +235,120 @@ def read_case(self, file_name: str, lightweight_mode: bool = False):
             fut.add_done_callback(functools.partial(Solver._sync_from_future, self))
         else:
             self.file.read(file_type="case", file_name=file_name)
+
```

Comment:
@hpohekar It will be a good exercise for you to extract and consolidate all the duplicated code.

Reply:
Right, I tried to add them in `BaseSession` to avoid all duplication but `tui` for meshing and `file` for solver is not available there.

Reply:
Yes sure. Thank you.